src/keboola/component/dao.py

```diff
@@ -426,7 +542,8 @@ def build_from_manifest(cls,
                             ):
         raise NotImplementedError
 
-    def _filter_attributes_by_manifest_type(self, manifest_type: Literal["in", "out"], legacy_queue: bool = False):
+    def _filter_attributes_by_manifest_type(self, manifest_type: Literal["in", "out"], legacy_queue: bool = False,
```

Comment:
Tady jsem si vubec nevsiml, ze je to uvnitr abstract classy. To je imo anti-pattern.  Nema smysl setupovat vsechny fieldy co jsou specificky jenom pro nektery druhy v abstract tride. Ty musi byt explicitne definovany na urovni implementacni tridy. Nastesti je ta metoda pouzita jenom v TableDefinition. **Tohle dokonce ohalilo bug v testech test_dao**`TestFileDefinition` je nestovany uvnitr `TestTableDefinition`, takze se nespousti jako FileDefinition. Kdyz to presunes tak to zacne kvuli temhle upravam padat.

Jestli to tady chces nechat tak muzes udelat abstract metodu co ti vrati ty fieldy `fields` a bude to implementovany na urovni kazdy specificky tridy (TableDefinition a FileDefinition). Tak zarucis ze se tam nebudou michat pojmy